dbus/src/message/parser.rs

```diff
@@ -0,0 +1,290 @@
+use std::error::Error as stdError;
+use arrayvec::ArrayVec;
+use std::fmt::{Debug, Formatter, Display};
+use crate::message::MatchRule;
+use crate::{MessageType, Path};
+use std::convert::TryFrom;
+use crate::strings::{Interface, BusName, Member};
+
+// Our grammar:
+// rules: rule (, rule)*
+// rule: sender | type | interface | member | path | path_namespace | destination | arg | arg_path
+// bool : 'true' | 'false'
+// type: "type" "=" message_type
+// message_type: "'signal'" | "'method_call'" | "'method_return'" | "'error'"
+// sender: "sender" "=" string
+// interface: "interface" "=" string
+// member: "member" "=" string
+// path: "path" "=" string
+// path_namespace: "path_namespace" "=" string
+// destination: "destination" "=" string
+// arg: "arg" 0-63 "=" string
+// arg_path: "arg" 0-63 "path" "=" string
+// eavesdrop: "eavesdrop" "=" bool
+
+/// The maximum rule elements supported by DBus
+/// This is taken from https://github.com/freedesktop/dbus/blob/master/bus/signals.c#L720
+const MAX_RULE_TOKENS: usize = 16;
+
+#[derive(Clone, Debug)]
+/// Error type that covers errors that might happen during parsing.
+pub enum Error {
+    /// DBus has a limit on the size of match rules.
```

Comment:
This needs to be publicly exported somehow, e g as `dbus::message::MatchRuleParserError` or something shorter.

Reply:
I think I've done this but I'm not 100% sure if it works like I want it to. Can you maybe double check?

Reply:
It looks as I expect it to. A tip if you want to double check is to run `cargo doc --open --no-deps` and check the `message` module to see if it shows up there.